src/Nancy/NancyEngine.cs

```diff
@@ -115,9 +115,16 @@ public Task<NancyContext> HandleRequest(Request request, Func<NancyContext, Nanc
             var staticContentResponse = this.staticContentProvider.GetContent(context);
             if (staticContentResponse != null)
             {
-                context.Response = staticContentResponse;
-                tcs.SetResult(context);
-                return tcs.Task;
+                try
+                {
+                    context.Response = staticContentResponse;
+                    tcs.SetResult(context);
+                    return tcs.Task;

```

Comment:
Does the TaskCompletionSource use or reference the Cancellation Token?  I don't quite understand how the 
`try ... finally { cts.Dispose() }` is any different than

    cts.Dispose(); 
    return tsc.Task;

This isn't dependent on the asynchronous execution of anything, is it?

Reply:
The only difference is that with `try{...}finally { cts.Dispose();}` will still dispose of the `CancellationTokenSource` whether an exception occurs on the intervening lines or not.

Reply:
Ah, going for the equivalent behavior of declaring a using block inside this scope, and there is some possibility of an exception setting the result or the response.  That makes more sense to me.

Reply:
In theory each of the individual `try...finally...` could be replaced with separate `using (cts) {...}` right?  (Just working out some personal edification here)

Reply:
In these cases, yes, I think it would be functionally the same. Personally, I like being more explicit that we're only disposing of the item here since the typical use of `using` also instantiates the object and could potentially be misinterpreted by others looking at the code.